### PR TITLE
Feature/products

### PR DESCRIPTION
**Pull request by Attila00**
PR for Product feature files and some common updation's.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
+import { Dialog } from './components/Pages/Modal/Dialog';
+import {Route, Switch, HashRouter as Router, Redirect} from 'react-router-dom';
 
 const App =()=>{
-    // For using locale in the application
-    const {t, i18n} = useTranslation();
+    // For using locale in the application. t is the core translation function, which through our key value passed
+    // as param fetches the related value from our languages objects file. 'i18n' holds many properties used 
+    // for getting current state of langugae in the application and updating it.
+    const {t} = useTranslation();
```
Comment (salilkapoor) [inline src/App.js:20]: where it's being used?

File: src/components/Organisms/header/Header.js [hunk 2]
```diff
@@ -3,30 +3,38 @@ import Icon from '../../atoms/Icon/Icon';
...
-    const {t, i18n} = useTranslation();
+    const {t} = useTranslation();
+    //Function to update selected as per clicked element
+    const updateActiveStatus = (e) =>{
+        document.querySelector('.selected') && 
+        document.querySelector('.selected').classList.remove('selected');
+        (e.target.tagName === "A") ? 
+        e.target.classList.add('selected') : 
+        (e.target.tagName === "IMG" && document.querySelector('#header__nav-home-item_home-link').classList.add('selected'));
+    };
```
Comment (salilkapoor) [inline src/components/Organisms/header/Header.js:18]: This is anti-pattern stuff in terms of React. You manipulating DOM directly.

File: src/components/Pages/Products/Products.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import React from 'react';
```
Comment (salilkapoor) [inline src/components/Pages/Products/Products.js:1]: Try offloading the load from this component.

File: src/components/Pages/Products/Products.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import Button from '../../atoms/Button/Button';
+import { useTranslation } from 'react-i18next';
+import { updateTitle } from '../../../utils/utils';
+import { scrollTop } from '../../../utils/scrollutils';
+import Scrolltop from '../../molecules/ScrollTop/Scrolltop';
+import { GlobalContext } from '../../../Context/globalContext';
+const Products = (props) =>{
+    const [totalProductsList] = React.useState(require('../../../../server/products/index.get.json'));
+    const [displayProductList, setDisplayProductList] = React.useState(totalProductsList);
+    const [selectedCategory, setSelectedCategory] = React.useState("");
```
Comment (salilkapoor) [inline src/components/Pages/Products/Products.js:14]: Too much complex state handling.

File: src/components/Pages/Products/Products.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            selected/>
+        </section>
+        <section id="products_list" className="products_list" tabIndex="0">
+            {displayProductList.length > 0 && displayProductList.map(product =>{
+            return <CardProduct details={product}/>
+        })}
+        </section>
+    </main>
+    )}
+export default Products;
```
Comment (salilkapoor) [inline src/components/Pages/Products/Products.js:73]: Looking at this, I got confused. There is Global context, then local state for this component, and then to children, you are passing props to. Let's discuss the thought process behind this.

File: src/components/atoms/Icon/Icon.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
 import './Icon.scss';
-export default function Image({source, height, width, alt, reqclass}){
-    return <img src={source} alt={alt} className={`${reqclass} `} height={height} width={width} />
+export default function Image({source, height, width, alt, reqclass, onClickHandler}){
```
Comment (salilkapoor) [inline src/components/atoms/Icon/Icon.js:3]: Why does Icon component need on click?

File: src/components/molecules/Card-Product/CardProduct.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    const {t} = useTranslation();
+    const addToCart = () =>{        
+        dispatch({
+            type:'ADD_ITEM',
+            payload:{...details, quantity:1, totalPrice:details.price}
+        });
+    }
+    return (
+        <>
+    <div className="product" tabIndex="0" role="card">
```
Comment (salilkapoor) [inline src/components/molecules/Card-Product/CardProduct.js:18]: Kindly share the thought. are you trying to make a card focusable element?

File: src/components/molecules/Card-Product/CardProduct.scss [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+            font-size: $font-medium;
+            @media screen and(max-width:$breakpoint-mobile){
+                height: auto;
+            }
+        }
+        &_card-image{
+            grid-column: 1 / 3;
+            grid-row: 2 ;
+            object-fit: contain ;
+            @media screen and(max-width:$breakpoint-tablet){
```
Comment (salilkapoor) [inline src/components/molecules/Card-Product/CardProduct.scss:26]: mobile first approach??

File: src/components/molecules/Productdropdown/Dropdown.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import Button from '../../atoms/Button/Button';
+import Icon from '../../atoms/Icon/Icon';
+import {useTranslation} from 'react-i18next';
+import '../../../assets/images/upward.png'
+import './Dropdown.scss';
+const Dropdown = ({list, headerTitle, selected, selectedCategory, itemClickHandler, displayProp}) =>{
+    const [isListOpen, setIsListOpen] = React.useState(false);
```
Comment (salilkapoor) [inline src/components/molecules/Productdropdown/Dropdown.js:8]: Let's discuss this component

**Review by salilkapoor — CHANGES_REQUESTED** [review]

File: src/components/atoms/Icon/Icon.js (hunk 6 above)
Comment (Attila00) [reply src/components/atoms/Icon/Icon.js:3]: In Footer component, I have added an image for changing application locale/language. For that purpose its added.

File: src/components/molecules/Card-Product/CardProduct.js (hunk 7 above)
Comment (Attila00) [reply src/components/molecules/Card-Product/CardProduct.js:18]: Yes Salil, complete product card on focus should also be detectable supposedly. We can surely discuss upon it.

File: src/components/Pages/Products/Products.js (hunk 5 above)
Comment (Attila00) [reply src/components/Pages/Products/Products.js:73]: Global context is for the 'Add to Card' functionality's state sharing between products page and cart page. Rest is for data sharing main products page and its sub components. Surely we shall discuss upon this, as per your guidance, we can fine tune components and state sharing.

File: src/App.js (hunk 1 above)
Comment (Attila00) [reply src/App.js:20]: It was mentioned in i18next docs, thus added it. I'll check by commenting it too.

File: src/components/molecules/Productdropdown/Dropdown.js (hunk 9 above)
Comment (Attila00) [reply src/components/molecules/Productdropdown/Dropdown.js:8]: Sure Salil.

File: src/components/Organisms/header/Header.js (hunk 2 above)
Comment (Attila00) [reply src/components/Organisms/header/Header.js:18]: Handled it through state and url updates now.

File: src/components/Pages/Products/Products.js (hunk 4 above)
Comment (Attila00) [reply src/components/Pages/Products/Products.js:14]: Upon discussion, will try to reduce the load a bit, as i will need some guidance regarding the same.

File: src/components/molecules/Card-Product/CardProduct.scss (hunk 8 above)
Comment (Attila00) [reply src/components/molecules/Card-Product/CardProduct.scss:26]: Tried to convert it to mobile first by giving default styles for mobile.

**Pull request merged**